ui/raidboss/popup-text.js

```diff
@@ -361,354 +370,410 @@ class PopupText {
-
-      // Text to speech overrides all other sounds.  This is so
-      // that a user who prefers tts can still get the benefit
-      // of infoText triggers without tts entries by turning
-      // on (speech=true, text=true, sound=true) but this will
-      // not cause tts to play over top of sounds or noises.
-      if (ttsText && playSpeech) {
-        // Heuristics for auto tts.
-        // * In case this is an integer.
-        ttsText = ttsText.toString();
-        // * Remove a bunch of chars.
-        ttsText = ttsText.replace(/[#!]/g, '');
-        // * slashes between mechanics
-        ttsText = ttsText.replace('/', ' ');
-        // * arrows helping visually simple to understand e.g. ↖ Front left / Back right ↘
-        ttsText = ttsText.replace(/[↖-↙]/g, '');
-        // * Korean TTS reads wrong with '1번째'
-        ttsText = ttsText.replace('1번째', '첫번째');
-        // * arrows at the front or the end are directions, e.g. "east =>"
-        ttsText = ttsText.replace(/[-=]>\s*$/g, '');
-        ttsText = ttsText.replace(/^\s*<[-=]/g, '');
-        // * arrows in the middle are a sequence, e.g. "in => out => spread"
-        let lang = this.options.AlertsLanguage || this.options.Language || 'en';
-        let arrowReplacement = {
-          en: ' then ',
-          cn: '然后',
-          de: ' dann ',
-          fr: ' puis ',
-          ja: 'や',
-          ko: ' 그리고 ',
-        };
-        ttsText = ttsText.replace(/\s*(<[-=]|[=-]>)\s*/g, arrowReplacement[lang]);
-        this.ttsSay(ttsText);
-      } else if (soundUrl && playSounds) {
-        let audio = new Audio(soundUrl);
-        audio.volume = soundVol;
-        audio.play();
-      }
+  _onTriggerInternalRun(triggerHelper) {
+    if ('run' in triggerHelper.trigger)
+      triggerHelper.trigger.run(this.data, triggerHelper.matches);
+  }
 
-      if ('run' in trigger)
-        trigger.run(this.data, matches);
-    };
+  _addText(container, e) {
+    container.appendChild(e);
+    if (container.children.length > this.kMaxRowsOfText)
+      container.removeChild(container.children[0]);
+  }
 
-    let promiseThenTrigger = () => {
-      // Put the resolution of the `promise` field inside this function so that
-      // It occurs after delaySeconds.
-      let promise = null;
-      if ('promise' in trigger) {
-        if (typeof trigger.promise === 'function') {
-          promise = trigger.promise(this.data, matches);
-          // Make sure we actually get a Promise back from the function
-          if (Promise.resolve(promise) !== promise) {
-            console.error('Trigger ' + trigger.id + ': promise function did not return a promise');
-            promise = null;
-          }
-        } else {
-          console.error('Trigger ' + trigger.id + ': promise defined but not a function');
+  _addTextFor(textType, triggerHelper) {
+    let textTypeUpper = textType[0].toUpperCase() + textType.slice(1);
+    // infoText
```

Comment:
Thanks for the comments! Can you add one for `// Info` on the top line?

Reply:
Done.